xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -252,6 +256,13 @@ public String toString()
         Pattern
             .compile("</?+(html|img|a|i|br?|embed|script|form|input|textarea|object|font|li|[dou]l|table|center|hr|p) ?([^>]*+)>");
 
+    /**
+     * Format for passing xproperties references in URLs. General format:
+     * {@code &lt;space&gt;.&lt;pageClass&gt;_&lt;number&gt;_&lt;propertyName&gt;}
+     * (e.g. {@code XWiki.XWikiRights_0_member}).
+     */
+    private static final Pattern XPROPERTY_URL_REFERENCE = Pattern.compile("(.+?)_([0-9]+)_(.+)");
```

Comment:
Should probably be named `XPROPERTY_REFERENCE_PATTERN`, for clarity, since the variable is not a EntityReference.